File: aosp/src/apps/audio_equalizer/app/src/main/java/com/leandromendes/audioequalizer/ui/MainViewModel.java
package com.leandromendes.audioequalizer.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.leandromendes.audioequalizer.data.model.EqualizerProfile;
import com.leandromendes.audioequalizer.data.repository.UserRepository;

import java.util.List;

public class MainViewModel extends ViewModel {

    UserRepository userRepository;

    private MutableLiveData<String> toastText = new MutableLiveData<>();

    public MainViewModel(){
        userRepository = new UserRepository();
    }

    public List<EqualizerProfile> GetAllEqualizerProfiles(){
        return userRepository.getAllEqualizerProfiles();
    }

    public EqualizerProfile GetEqualizerProfiles(int position){
        return userRepository.getEqualizerProfile(position);
    }

    public void AddProfile(EqualizerProfile equalizerProfile){
        userRepository.AddProfile(equalizerProfile);
    }

    public void UpdateProfile(int position, EqualizerProfile equalizerProfile){
        userRepository.UpdateProfile(position, equalizerProfile);
    }

    public void RemoveProfile(int position){
        userRepository.RemoveProfile(position);
    }

    public LiveData<String> getToastText() {
        return toastText;
    }

    public void SetToastText(String text) {
        toastText.setValue(text);
    }
}

File: aosp/src/apps/VehicleEqualizerFull/app/src/main/java/com/leandromendes/vehicleequalizer/ui/viewmodel/MainViewModel.kt
package com.leandromendes.vehicleequalizer.ui.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.leandromendes.vehicleequalizer.data.model.EqualizerProfile
import com.leandromendes.vehicleequalizer.data.repository.UserRepository
import kotlinx.coroutines.launch

// MainViewModel now receives the Repository in the constructor
class MainViewModel(private val userRepository: UserRepository) : ViewModel() {

    // LiveData that MainActivity will OBSERVE
    val allProfilesLiveData: LiveData<List<EqualizerProfile>> = userRepository.allEqualizerProfiles

    private val toastText = MutableLiveData<String?>()

    fun addProfile(equalizerProfile: EqualizerProfile) = viewModelScope.launch {
        userRepository.addProfile(equalizerProfile)
    }

    fun updateProfile(equalizerProfile: EqualizerProfile) = viewModelScope.launch {
        userRepository.updateProfile(equalizerProfile)
    }

    fun removeProfile(equalizerProfile: EqualizerProfile) = viewModelScope.launch {
        userRepository.removeProfile(equalizerProfile)
    }

    fun getToastText(): LiveData<String?> {
        return toastText
    }

    fun setToastText(text: String?) {
        toastText.value = text
    }
}